public/app.js

```diff
@@ -0,0 +1,380 @@
+
+      let elements = document.querySelectorAll('#box')
+
+      elements.forEach(element => {
+
+        element.insertAdjacentHTML('beforeend', createNewElement(type, name, img, artist));
+
+      });
+    },
+    /**
+     * @description Функция очистки элемента песни в html
+     */
+    resetTrackDetail() {
+      this.inputField().songDetail.innerHTML = '';
+    },
+    /**
+     * @description Функция очистки списка песен
+     */
+    resetTracks() {
+      this.inputField().tracks.innerHTML = '';
+      this.resetTrackDetail();
+    },
+    /**
+     * @description Функция очистки списка плейлистов
+     */
+    resetPlaylist() {
+      this.inputField().playlist.innerHTML = '';
+      this.resetTracks();
+    },
+    /**
+     * @description Функция сохранения токена
+     */
+    storeToken(value) {
+      document.querySelector(DOMElements.hfToken).value = value;
+    },
+    /**
+     * @description Функция получения сохраненного токена
+     */
+    getStoredToken() {
+      return {
+        token: document.querySelector(DOMElements.hfToken).value
+      }
+    }
+  }
+
+})();
+
+const APPController = (function(UICtrl, APICtrl) {
+
+  const DOMInputs = UICtrl.inputField();
+  /**
+   * @description Функция загрузки жанров при загрузке страницы
+   */
+  const loadGenres = async () => {
+
+    const token = await APICtrl.getToken();
+
+    UICtrl.storeToken(token);
+
+    const genres = await APICtrl.getGenres(token);
+
+    genres.forEach(element => UICtrl.createGenre(element.name, element.id));
+  }
+
+  const getToken = async () => {
+
+    const token = await APICtrl.getToken();
+
+    UICtrl.storeToken(token);
+
+    return token;
+  }
+
+  /**
+   * @description Функция загрузки плейлистов при загрузке страницы
+   */
+  const loadPlaylists = async () => {
+
+    const token = await APICtrl.getToken();
+
+    const playlist = await APICtrl.getPlaylist(token, '3PZTFPQhr0vHnYGwFUvQco');
+
+    UICtrl.createDetail('playlist', playlist.name, playlist.images[0].url, playlist.artists[0].name);
+  }
+
+  /**
+   * @description Функция загрузки песен при загрузке страницы
+   */
+  const loadSongs = async () => {
+
+    const token = await APICtrl.getToken();
+
+    const track = await APICtrl.getTrack(token, 'https://api.spotify.com/v1/tracks/1rDQ4oMwGJI7B4tovsBOxc');
+
+    UICtrl.createDetail('album', track.album.images[1].url, track.name, track.artists[0].name);
+  }
+
+  /**
+   * @description Функция создания <option> с подходящими плейлистами при выранном жанре
+   */
+  DOMInputs.genre.addEventListener('change', async () => {
+
+    UICtrl.resetPlaylist();
+
+    const token = UICtrl.getStoredToken().token;
+
+    const genreSelect = UICtrl.inputField().genre;
+
+    const genreId = genreSelect.options[genreSelect.selectedIndex].value;
+
+    const playlist = await APICtrl.getPlaylistByGenre(token, genreId);
+
+    playlist.forEach(p => UICtrl.createPlaylist(p.name, p.tracks.href));
```

Comment:
Тут наглядный пример того, что переменная названа не корректно.

Если в переменной находится массив, то переменная должна говорить о том, что в ней не просто значение, а массив из значений.
Например
sound = ['sound1', 'sound2'];
sounds= ['sound1', 'sound2'];

В первом случае, при использовании переменой, будет неожиданным, что мы работаем с массивом. Во 2 примере, не смотря на значение в переменной, мы понимаем что работаем с массивом.


Тут должно быть как-то так
```suggestion
    playlists.forEach(playlist => UICtrl.createPlaylist(playlist.name, playlist.tracks.href));
```